dvc/remote/gdrive.py

```diff
@@ -84,16 +84,51 @@ def __init__(self, repo, config):
         self.init_drive()
 
     def init_drive(self):
-        self.client_id = self.config.get(Config.SECTION_GDRIVE_CLIENT_ID, None)
-        self.client_secret = self.config.get(
-            Config.SECTION_GDRIVE_CLIENT_SECRET, None
+        self.use_service_account = False
+        self.service_account_email = self.config.get(
+            Config.SECTION_GDRIVE_SERVICE_ACCOUNT_EMAIL, None
         )
-        if not self.client_id or not self.client_secret:
+        self.service_account_user_email = self.config.get(
+            Config.SECTION_GDRIVE_SERVICE_ACCOUNT_USER_EMAIL, None
+        )
+        self.service_account_p12_file_path = self.config.get(
+            Config.SECTION_GDRIVE_SERVICE_ACCOUNT_P12_FILE_PATH, None
+        )
+        if (
+            self.service_account_email
+            and self.service_account_user_email
+            and self.service_account_p12_file_path
+        ):
+            self.use_service_account = True
```

Comment:
let's make it an explicit option? It'll simplify all this logic. Also users will be able to switch between different credentials if needed.

Reply:
Have added DVC config option `gdrive_user_service_account`